### PR TITLE
docs: add a brief components doc

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/test-components-js.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+npm i @playwright/experimental-ct-react
+# npm i @playwright/experimental-ct-vue
+# npm i @playwright/experimental-ct-svelte
+```
+
+### Step 2: create `playwright/index.html`
+```html
+<html lang="en">
+  <body>
+    <div id="root"></div>
```
Comment (aslushnikov) [inline docs/src/test-components-js.md:51]:
```suggestion
```

File: docs/src/test-components-js.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+npx playwright test
+```
+
+### Further reading: configure reporting, browsers, tracing
+
+Refer to [Playwright config](./test-configuration.md) for configuring your project.
+
+## Under the hood
+
+When Playwright Test is used to test web components, tests run in Node.js, while components run in the real browser. This brings together the best of both worlds: component run in the real browser environment, real clicks are triggered, real layout is executed, visual regression is possible. At the same time, test can use all the powers of Node.js as well as all the Playwright Test features. As a result, the same parallel, parametrized tests with the same post-mortem Tracing story are available during component testing.
```
Comment (aslushnikov) [inline docs/src/test-components-js.md:88]:
```suggestion
When Playwright Test is used to test web components, tests run in Node.js, while components run in the real browser. This brings together the best of both worlds: components run in the real browser environment, real clicks are triggered, real layout is executed, visual regression is possible. At the same time, test can use all the powers of Node.js as well as all the Playwright Test features. As a result, the same parallel, parametrized tests with the same post-mortem Tracing story are available during component testing.
```

File: docs/src/test-components-js.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+Here is how this is achieved:
+
+- Once the tests are executed, Playwright creates a list of components that the tests need.
+- It then compiles a bundle that includes these components and serves it using a local static web server.
+- Upon the `mount` call within the test, Playwright navigates to the facade page of this bundle and tells it to render the component.
```
Comment (aslushnikov) [inline docs/src/test-components-js.md:94]:
```suggestion
- Upon the `mount` call within the test, Playwright navigates to the facade page `/playwright/index.html` of this bundle and tells it to render the component.
```

File: docs/src/test-components-js.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+- Events are marshalled back the the Node.js environment to allow verification.
```
Comment (aslushnikov) [inline docs/src/test-components-js.md:95]:
```suggestion
- Events are marshalled back to the Node.js environment to allow verification.
```

**Review by aslushnikov — APPROVED** [review]

File: docs/src/test-components-js.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+### Step 2: create `playwright/index.html`
+```html
+<html lang="en">
+  <body>
+    <div id="root"></div>
+    <script type="module" src="/playwright/index.ts"></script>
+  </body>
+</html>
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply docs/src/test-components-js.md:54]: I actually think it should be verbose for understanding of what it is...

File: docs/src/test-components-js.md (hunk 1 above)
Comment (pavelfeldman) [reply docs/src/test-components-js.md:51]: They should know they can put it into a container for testing

File: docs/src/test-components-js.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+// Apply theme here, add anything your component needs at runtime here.
+```
+
+### Create a test `src/App.spec.tsx`
+
+```js
+import { test, expect } from '@playwright/experimental-ct-react';
+import App from './App';
+
+test.use({ viewport: { width: 500, height: 500 } });
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply docs/src/test-components-js.md:68]: I'm just braggin!

**Pull request merged**